src/org/pentaho/reporting/platform/plugin/SimpleReportingAction.java

```diff
@@ -880,8 +880,10 @@ public boolean _execute() throws Exception {
       }
     }
     // SP-307, BISERVER-9688
-
     final MasterReport report = getReport();
+    // Set unlimited queryLimit for reports running in "Run in background mode" in PUC.
+    report.setQueryLimit( -1 );
```

Comment:
Don't delete the query limit. A user can set this limit on the master-report and would expect that limit to be honoured. So a blanked "make this report unlimited" operation like this is wrong. The report will be a freshly loaded report anyway, so actually: Dont do anything. Delete this row and the report will behave as it should.